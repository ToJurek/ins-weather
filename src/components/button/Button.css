.refresh-button {
    background-color: hsla(0, 0%, 5%, 0.71);
    border-radius: 14px;
    margin-top: 5%;
    width: 90%;
    border: 1px solid #3c3c3c;
    display: inline-block;
    cursor: pointer;
    color: #f0f5f5;
    font-size: 17px;
    padding: 16px 31px;
    text-decoration: none;
    text-shadow: 0px 1px 0px #f0f5f5;
    outline: none;
    font-weight: bold;
}

.refresh-button:hover {
    background-color: #f0f5f5;
    color: black;
}

.refresh-button:active {
    position: relative;
    top: 1px;
}
