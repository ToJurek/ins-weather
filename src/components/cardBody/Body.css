.last-update {
    font-size: 0.9em;
    margin-top: 3%;
}

.parameter-icon {
    margin-right: 10px;
}

.city {
    height: 30px;
}

.column {
    float: left;
    width: 50%;
}

.row:after {
    content: "";
    display: table;
    clear: both;
}

.column .temp {
    font-size: 3.5em;
}
