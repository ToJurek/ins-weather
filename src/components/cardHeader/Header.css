.header {
    font-size: 2em;
}

.map-marker {
    margin-right: 10px;
}
