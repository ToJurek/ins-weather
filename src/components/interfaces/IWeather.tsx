export default interface IWeather {
    icon: string;
    description: string;
    temp: string;
    feelsLike: string;
    tempMin: string;
    tempMax: string;
    pressure: string;
    humidity: string;
    clouds: string;
    time: string
}
