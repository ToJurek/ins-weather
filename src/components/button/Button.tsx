import React from 'react';
import './Button.css';

interface Props {
    name: string
    handleClick: any
}

function Button(props: Props) {
    return (<button className={'refresh-button'} onClick={() => props.handleClick()}>{props.name}</button>)
}

export default Button
