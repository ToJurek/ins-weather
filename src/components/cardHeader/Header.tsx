import React from 'react';
import './Header.css';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faMapMarkerAlt} from "@fortawesome/free-solid-svg-icons";

interface Props {
    city: string
}

function Header(props: Props) {
    return (
        <div className={'header'}>
            <FontAwesomeIcon icon={faMapMarkerAlt} className={'map-marker'}/>
            <span className={'city'}>{props.city}</span>
        </div>
    )
}

export default Header;
