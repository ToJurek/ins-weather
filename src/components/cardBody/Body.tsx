import React from 'react';
import './Body.css';
import IWeather from "../interfaces/IWeather";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faAngleDoubleDown, faCloudSun, faTint} from "@fortawesome/free-solid-svg-icons";
import {IconProp} from "@fortawesome/fontawesome-svg-core";

interface Parameter {
    icon: IconProp,
    value: string
}

interface Props {
    weather: IWeather
}

function Body(props: Props) {
    let weather = props.weather;
    let parameters: Parameter[] = [{icon: faCloudSun, value: weather.clouds}, {
        icon: faTint,
        value: weather.humidity
    }, {icon: faAngleDoubleDown, value: weather.pressure}];
    return (
        <>
            <div className={'last-update'}>Last update: {weather.time}</div>
            <div className={'row'}>
                <div className={'column'}><img
                    src={'http://openweathermap.org/img/wn/' + weather.icon + '@2x.png'}/></div>
                <div className={'column'}><span className={'temp'}>{weather.temp}</span></div>
            </div>
            <div>{weather.tempMax + '/' + weather.tempMin + ' RealFeel ' + weather.feelsLike}</div>
            <div>{weather.description}</div>
            {renderParameters(parameters)}
        </>
    )

}

function renderParameters(parameters: Parameter[]): React.ReactNode | null {
    return parameters.map(parameter => {
        return (<div>
            <FontAwesomeIcon icon={parameter.icon} className={'parameter-icon'}/>
            <span>{parameter.value}</span>
        </div>)
    })
}

export default Body;
