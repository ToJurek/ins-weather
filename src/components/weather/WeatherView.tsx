import React from 'react';
import './Weather.css';
import IWeather from "../interfaces/IWeather";
import Header from "../cardHeader/Header";
import Body from "../cardBody/Body";
import Button from "../button/Button";

interface Props {
    weather: IWeather,
    refetchData: any
}

function WeatherView(props: Props) {
    let weather: IWeather = props.weather;
    return (
        <div className={'content'}>
            <div className={"dashboard"}>
                <Header city={"WrocÅ‚aw"}/>
                <Body weather={weather}/>
            </div>
            <Button handleClick={props.refetchData} name={"Update"}/>
        </div>
    )

}

export default WeatherView;
