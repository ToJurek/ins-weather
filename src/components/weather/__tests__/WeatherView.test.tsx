import React from "react";
import Enzyme, {mount} from 'enzyme';
import WeatherView from "../WeatherView";
import Button from "../../button/Button";
import Header from "../../cardHeader/Header";
import Body from "../../cardBody/Body";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import IWeather from "../../interfaces/IWeather";

Enzyme.configure({adapter: new Adapter()});

describe('WeatherView', function () {
    let wrapper: any
    let mockFunction: any
    const weather: IWeather = {
        icon: '04d',
        description: 'desc',
        temp: '13',
        feelsLike: '12',
        tempMin: '11',
        tempMax: '15',
        pressure: '1000 hPa',
        humidity: '12%',
        clouds: '10%',
        time: '12/02/1992 12:11'
    }
    beforeEach(() => {
        mockFunction = jest.fn()
        wrapper = mount(<WeatherView weather={weather} refetchData={mockFunction}/>)
    })

    test('should match to snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('should call children', () => {
        expect(wrapper.find(Header).exists()).toBeTruthy()
        expect(wrapper.find(Body).exists()).toBeTruthy()
        expect(wrapper.find(Button).exists()).toBeTruthy()
    })

    test('should render button name correctly', () => {
        expect(wrapper.find('button').text()).toBe('Update')
    })

    test('should mock function be called', () => {
        wrapper.find(Button).simulate('click');
        expect(mockFunction.mock.calls.length).toEqual(1)
    })
});
