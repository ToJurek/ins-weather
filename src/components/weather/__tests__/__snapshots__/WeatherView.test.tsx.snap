// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherView should match to snapshot 1`] = `
<WeatherView
  refetchData={[MockFunction]}
  weather={
    Object {
      "clouds": "10%",
      "description": "desc",
      "feelsLike": "12",
      "humidity": "12%",
      "icon": "04d",
      "pressure": "1000 hPa",
      "temp": "13",
      "tempMax": "15",
      "tempMin": "11",
      "time": "12/02/1992 12:11",
    }
  }
>
  <div
    className="content"
  >
    <div
      className="dashboard"
    >
      <Header
        city="Wrocław"
      >
        <div
          className="header"
        >
          <FontAwesomeIcon
            border={false}
            className="map-marker"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  384,
                  512,
                  Array [],
                  "f3c5",
                  "M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z",
                ],
                "iconName": "map-marker-alt",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-map-marker-alt fa-w-12 map-marker"
              data-icon="map-marker-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
          <span
            className="city"
          >
            Wrocław
          </span>
        </div>
      </Header>
      <Body
        weather={
          Object {
            "clouds": "10%",
            "description": "desc",
            "feelsLike": "12",
            "humidity": "12%",
            "icon": "04d",
            "pressure": "1000 hPa",
            "temp": "13",
            "tempMax": "15",
            "tempMin": "11",
            "time": "12/02/1992 12:11",
          }
        }
      >
        <div
          className="last-update"
        >
          Last update: 
          12/02/1992 12:11
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <img
              alt="weather-icon"
              src="http://openweathermap.org/img/wn/04d@2x.png"
            />
          </div>
          <div
            className="column"
          >
            <span
              className="temp"
            >
              13
            </span>
          </div>
        </div>
        <div>
          15/11 RealFeel 12
        </div>
        <div>
          desc
        </div>
        <div
          key="0"
        >
          <FontAwesomeIcon
            border={false}
            className="parameter-icon"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  640,
                  512,
                  Array [],
                  "f6c4",
                  "M575.2 325.7c.2-1.9.8-3.7.8-5.6 0-35.3-28.7-64-64-64-12.6 0-24.2 3.8-34.1 10-17.6-38.8-56.5-66-101.9-66-61.8 0-112 50.1-112 112 0 3 .7 5.8.9 8.7-49.6 3.7-88.9 44.7-88.9 95.3 0 53 43 96 96 96h272c53 0 96-43 96-96 0-42.1-27.2-77.4-64.8-90.4zm-430.4-22.6c-43.7-43.7-43.7-114.7 0-158.3 43.7-43.7 114.7-43.7 158.4 0 9.7 9.7 16.9 20.9 22.3 32.7 9.8-3.7 20.1-6 30.7-7.5L386 81.1c4-11.9-7.3-23.1-19.2-19.2L279 91.2 237.5 8.4C232-2.8 216-2.8 210.4 8.4L169 91.2 81.1 61.9C69.3 58 58 69.3 61.9 81.1l29.3 87.8-82.8 41.5c-11.2 5.6-11.2 21.5 0 27.1l82.8 41.4-29.3 87.8c-4 11.9 7.3 23.1 19.2 19.2l76.1-25.3c6.1-12.4 14-23.7 23.6-33.5-13.1-5.4-25.4-13.4-36-24zm-4.8-79.2c0 40.8 29.3 74.8 67.9 82.3 8-4.7 16.3-8.8 25.2-11.7 5.4-44.3 31-82.5 67.4-105C287.3 160.4 258 140 224 140c-46.3 0-84 37.6-84 83.9z",
                ],
                "iconName": "cloud-sun",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-cloud-sun fa-w-20 parameter-icon"
              data-icon="cloud-sun"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M575.2 325.7c.2-1.9.8-3.7.8-5.6 0-35.3-28.7-64-64-64-12.6 0-24.2 3.8-34.1 10-17.6-38.8-56.5-66-101.9-66-61.8 0-112 50.1-112 112 0 3 .7 5.8.9 8.7-49.6 3.7-88.9 44.7-88.9 95.3 0 53 43 96 96 96h272c53 0 96-43 96-96 0-42.1-27.2-77.4-64.8-90.4zm-430.4-22.6c-43.7-43.7-43.7-114.7 0-158.3 43.7-43.7 114.7-43.7 158.4 0 9.7 9.7 16.9 20.9 22.3 32.7 9.8-3.7 20.1-6 30.7-7.5L386 81.1c4-11.9-7.3-23.1-19.2-19.2L279 91.2 237.5 8.4C232-2.8 216-2.8 210.4 8.4L169 91.2 81.1 61.9C69.3 58 58 69.3 61.9 81.1l29.3 87.8-82.8 41.5c-11.2 5.6-11.2 21.5 0 27.1l82.8 41.4-29.3 87.8c-4 11.9 7.3 23.1 19.2 19.2l76.1-25.3c6.1-12.4 14-23.7 23.6-33.5-13.1-5.4-25.4-13.4-36-24zm-4.8-79.2c0 40.8 29.3 74.8 67.9 82.3 8-4.7 16.3-8.8 25.2-11.7 5.4-44.3 31-82.5 67.4-105C287.3 160.4 258 140 224 140c-46.3 0-84 37.6-84 83.9z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
          <span>
            10%
          </span>
        </div>
        <div
          key="1"
        >
          <FontAwesomeIcon
            border={false}
            className="parameter-icon"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  352,
                  512,
                  Array [],
                  "f043",
                  "M205.22 22.09c-7.94-28.78-49.44-30.12-58.44 0C100.01 179.85 0 222.72 0 333.91 0 432.35 78.72 512 176 512s176-79.65 176-178.09c0-111.75-99.79-153.34-146.78-311.82zM176 448c-61.75 0-112-50.25-112-112 0-8.84 7.16-16 16-16s16 7.16 16 16c0 44.11 35.89 80 80 80 8.84 0 16 7.16 16 16s-7.16 16-16 16z",
                ],
                "iconName": "tint",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-tint fa-w-11 parameter-icon"
              data-icon="tint"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M205.22 22.09c-7.94-28.78-49.44-30.12-58.44 0C100.01 179.85 0 222.72 0 333.91 0 432.35 78.72 512 176 512s176-79.65 176-178.09c0-111.75-99.79-153.34-146.78-311.82zM176 448c-61.75 0-112-50.25-112-112 0-8.84 7.16-16 16-16s16 7.16 16 16c0 44.11 35.89 80 80 80 8.84 0 16 7.16 16 16s-7.16 16-16 16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
          <span>
            12%
          </span>
        </div>
        <div
          key="2"
        >
          <FontAwesomeIcon
            border={false}
            className="parameter-icon"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  320,
                  512,
                  Array [],
                  "f103",
                  "M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z",
                ],
                "iconName": "angle-double-down",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-angle-double-down fa-w-10 parameter-icon"
              data-icon="angle-double-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
          <span>
            1000 hPa
          </span>
        </div>
      </Body>
    </div>
    <Button
      handleClick={[MockFunction]}
      name="Update"
    >
      <button
        className="refresh-button"
        onClick={[Function]}
      >
        Update
      </button>
    </Button>
  </div>
</WeatherView>
`;
